rsts/deployment/configuration/auth_setup.rst

```diff
@@ -51,178 +41,346 @@ Authentication Setup
+      .. prompt:: bash $
 
-Save and exit your editor.
+         kubectl rollout restart deployment/flyteadmin -n flyte
 
-Restart `flyteadmin` for the changes to take effect:
+      10. Restart ``flytepropeller`` to start using authenticated requests:
 
-.. prompt:: bash $
+      .. prompt:: bash $
 
-   kubectl rollout restart deployment/flyteadmin -n flyte
+          kubectl rollout restart deployment/flytepropeller -n flyte
 
-Restart `flytepropeller` to start using authenticated requests:
+      11. Restart ``flytescheduler``` to start using authenticated requests:
 
-.. prompt:: bash $
+      .. prompt:: bash $
 
-   kubectl rollout restart deployment/flytepropeller -n flyte
+         kubectl rollout restart deployment/flytescheduler -n flyte
 
-Restart ``flytescheduler``` to start using authenticated requests:
+      12. For flytectl/pyflyte, make sure that your local config file (``$HOME/.flyte/config.yaml``) includes the following option:
 
-.. prompt:: bash $
+      .. code-block:: yaml
 
-   kubectl rollout restart deployment/flytescheduler -n flyte
+         admin:
+           ...
+           authType: Pkce #change from the default `clientCred` to enable client auth without using shared secrets
+           ...    
 
 .. note::
 
    **Congratulations!**
 
-   It should now be possible to go to flyte UI (https://<your domain>/console) and be prompted for authentication. Flytectl should automatically pickup the change and start prompting for authentication as well.
+   It should now be possible to go to Flyte UI and be prompted for authentication. Flytectl should automatically pickup the change and start prompting for authentication as well.
    If you want to use an external OAuth2 provider for App authentication, please continue reading into the next section.
 
-OAuth2 Authorization Server
----------------------------
+***************************
+Custom Authorization Server
+***************************
 
-An OAuth2 Authorization Server allows external clients to request to authenticate and act on behalf of users (or as their own identities). Having
-an OAuth2 Authorization Server enables Flyte administrators control over which apps can be installed and what scopes they are allowed to request or be granted (i.e. what privileges can they assume).
 
-Flyte comes with a built-in authorization server that can be statically configured with a set of clients to request and act on behalf of the user.
-The default clients are defined `here <https://github.com/flyteorg/flyteadmin/pull/168/files#diff-1267ff8bd9146e1c0ff22a9e9d53cfc56d71c1d47fed9905f95ed4bddf930f8eR74-R100>`__
-and the corresponding section can be modified through configs.
+Flyte ships with an internal authorization server, hence setting up an external Authorization Server is optional but recommended. Having an external Authorization Server enables Flyte administrators control over which apps can be installed and what privileges can be granted to users. 
```

Comment:
this isn't true... having an external auth server does not enable Flyte administrators to control over which apps can be installed.  In any case, what do you mean by an app here?  Apps aren't installable in Flyte at all and this auth server is not the rbac that people often ask about ('i want to limit some users to only be able to access certain flyte projects').  Maybe just truncate this line to after "is optional".

Reply:
That line is on the current version of the docs ([here](https://docs.flyte.org/en/latest/deployment/configuration/auth_setup.html#oauth2-authorization-server)) but agree that it can be misleading. Nevertheless, the docs should provide some reasoning on what are the reasons/situations when using an external auth server is preferred.